### PR TITLE
fix memory leak when request is aborted

### PR DESCRIPTION
**Pull request by jfromaniello**
I can't write a test case for this you will have to believe me. 😀

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -62,7 +62,13 @@ function setupRateLimitEventExt(server, options) {
         }
 
         const rlRequest = new RateLimitedRequest(request.id, resp);
-        const r = pendingRequests.keepWorse(rlRequest);
+        pendingRequests.keepWorse(rlRequest);
+
+        if (request.raw && request.raw.req) {
```
Comment (dschenkelman) [inline lib/index.js]: can we add a test for this case 🙏 ?

File: lib/index.js [hunk 2]
```diff
@@ -62,7 +62,13 @@ function setupRateLimitEventExt(server, options) {
         }
 
         const rlRequest = new RateLimitedRequest(request.id, resp);
-        const r = pendingRequests.keepWorse(rlRequest);
+        pendingRequests.keepWorse(rlRequest);
```
Comment (dschenkelman) [inline lib/index.js]: was `r` not being used?

Comment (jfromaniello) [inline lib/index.js]: it was not

Comment (dschenkelman) [inline lib/index.js]: 👍

**Pull request merged**